// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

message RuntimeconfigBetaVariable {
  string name = 1;
  string runtime_config = 2;
  string text = 3;
  string value = 4;
  string update_time = 5;
  string project = 6;
}

message ApplyRuntimeconfigBetaVariableRequest {
  RuntimeconfigBetaVariable resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteRuntimeconfigBetaVariableRequest {
  string service_account_file = 1;
  RuntimeconfigBetaVariable resource = 2;
}

message ListRuntimeconfigBetaVariableRequest {
  string service_account_file = 1;
  string Project = 2;
  string RuntimeConfig = 3;
}

message ListRuntimeconfigBetaVariableResponse {
  repeated RuntimeconfigBetaVariable items = 1;
}

service RuntimeconfigBetaVariableService {
  rpc ApplyRuntimeconfigBetaVariable(ApplyRuntimeconfigBetaVariableRequest) returns (RuntimeconfigBetaVariable);
  rpc DeleteRuntimeconfigBetaVariable(DeleteRuntimeconfigBetaVariableRequest) returns (google.protobuf.Empty);
  rpc ListRuntimeconfigBetaVariable(ListRuntimeconfigBetaVariableRequest) returns (ListRuntimeconfigBetaVariableResponse);
}
